topology/topology.py

```diff
@@ -0,0 +1,151 @@
+        node = used_slice.add_node(name=node_name, site=site)
+        node.set_capacities(cores=cores, ram=ram, disk=disk)
+        node.set_image(image)
+        self.nodes[node_name] = node
+        self.iface_table[node_name] = []
+
+    def add_interface(self, node_name: str, nic_name: str) -> None:
+        """Add a network interface to a node."""
+        node = self.nodes[node_name]
+        iface = node.add_component(model='NIC_Basic', name=nic_name).get_interfaces()[0]
+        self.iface_table[node_name].append(iface)
+
+    def create_network_connection(
+            self,
+            selected_slice,
+            network_name: str,
+            node1: str,
+            node2: str,
+            interface1_idx: int,
+            interface2_idx: int
+    ) -> None:
+        """Create a network connection between two nodes using specified interfaces."""
+        created_connections = []
+        if (node1, node2) not in created_connections and (node2, node1) not in created_connections:
+            selected_slice.add_l2network(
+                name=network_name,
+                interfaces=[
+                    self.iface_table[node1][interface1_idx],
+                    self.iface_table[node2][interface2_idx]
+                ]
+            )
+            created_connections.append((node1, node2))
+            print(f"\n > Created connection: {network_name} between {node1} and {node2}")
+
+    def setup_network(self, selected_slice, base_network_name: str, nodes: List[str]) -> None:
+        """Create a full mesh network between the specified nodes."""
+
+        # Flatten the connections into a list of tuples first
+        flat_connections = []
+        for conn_dict in self.connections:
+            for source, targets in conn_dict.items():
+                flat_connections.extend((source, target) for target in targets)
+
+        interface_indices = {node: 0 for node in nodes}
+
+        for i, (source_node, target_node) in enumerate(flat_connections):
+
+            network_name = f"{base_network_name}{i}"
+
+            # Create connection using current interface indices
+            self.create_network_connection(
+                selected_slice,
+                network_name,
+                source_node,
+                target_node,
+                interface_indices[source_node],
+                interface_indices[target_node]
+            )
+
+            # Increment interface indices
+            interface_indices[source_node] += 1
+            interface_indices[target_node] += 1
+
+    def setup_client_connections(
+            self,
+            selected_slice,
+            base_network_name: str,
+            client_connections: List[Tuple[str, str]]
+    ) -> None:
+        """Set up connections between client nodes and their designated targets."""
+        for idx, (client, target) in enumerate(client_connections):
+            network_name = f"{base_network_name}{idx}"
```

Comment:
need to unpack dictionary : 
for idx, connection in enumerate(client_connections):
    client, target = list(connection.items())[0]  
    print(f"Index: {idx}, Client: {client}, Target: {target}")